<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Vignette for the hancock package • hancock</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Vignette for the hancock package">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hancock</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/hancock.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/concepts.html">Concepts on cell types and signatures</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/kevinrue/hancock">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Vignette for the hancock package</h1>
                        <h4 class="author">Kevin Rue-Albrecht</h4>
            <address class="author_afil">
      Kennedy Institute of Rheumatology, University of Oxford, Headington, Oxford OX3 7FY, UK.<br><a class="author_email" href="mailto:#"></a><a href="mailto:kevin.rue-albrecht@kennedy.ox.ac.uk">kevin.rue-albrecht@kennedy.ox.ac.uk</a>
      </address>
                              <h4 class="author">Second Author</h4>
            <address class="author_afil">
      Second Author’s Affiliation<br><a class="author_email" href="mailto:#"></a><a href="mailto:corresponding@author.com">corresponding@author.com</a>
      </address>
                  
            <h4 class="date">15 February 2019</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/kevinrue/hancock/blob/master/vignettes/hancock.Rmd"><code>vignettes/hancock.Rmd</code></a></small>
      <div class="hidden name"><code>hancock.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      Example usage of the hancock package demonstrated using example data sets.
    </div>
    
<p><strong>Compiled date</strong>: 2019-02-15</p>
<p><strong>Last edited</strong>: 2018-03-08</p>
<p><strong>License</strong>: MIT + file LICENSE</p>
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>The goal of the <em><a href="https://github.com/kevinrue/hancock">hancock</a></em> package is to provide a collection of methods for learning and applying gene signatures associated with cellular phenotypes and identities. Particular focus is given to single-cell data stored in objects derived from the <em><a href="https://bioconductor.org/packages/3.9/SummarizedExperiment">SummarizedExperiment</a></em> class.</p>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<div id="setup" class="section level2">
<h2 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h2>
<p>To run an analysis, the first step is to start R and load the <em><a href="https://github.com/kevinrue/hancock">hancock</a></em> package:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(hancock)</a></code></pre></div>
</div>
<div id="example-data-set" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data-set" class="anchor"></a>Example data set</h2>
<p>As an initial example, we demonstrate an analysis to find genes markers of three populations of neurons labelled by the fluorescent protein tdTomato (tdT) after Cre-mediated recombination.</p>
<p>For this example, we use a subset of single-cell RNA-seq dataset from the Allen Brain Atlas <span class="citation">(Tasic et al. 2016)</span>. This dataset is available in the <em><a href="https://bioconductor.org/packages/3.9/scRNAseq">scRNAseq</a></em> experiment data package:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scRNAseq)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(allen)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">allen</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co">#&gt; class: SummarizedExperiment </span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co">#&gt; dim: 20908 379 </span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co">#&gt; metadata(2): SuppInfo which_qc</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co">#&gt; assays(4): tophat_counts cufflinks_fpkm rsem_counts rsem_tpm</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co">#&gt; rownames(20908): 0610007P14Rik 0610009B22Rik ... Zzef1 Zzz3</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="co">#&gt; rowData names(0):</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="co">#&gt; colnames(379): SRR2140028 SRR2140022 ... SRR2139341 SRR2139336</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="co">#&gt; colData names(22): NREADS NALIGNED ... Animal.ID</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="co">#&gt;   passes_qc_checks_s</span></a></code></pre></div>
<p>The individual Cre line is indicated for each cell in the <code>"driver_1_s"</code> metadata.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(allen<span class="op">$</span>driver_<span class="dv">1</span>_s)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; Ntsr1-Cre_GN220  Rbp4-Cre_KL100  Scnn1a-Tg3-Cre </span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt;              79             190             110</span></a></code></pre></div>
</div>
<div id="learning-signatures" class="section level2">
<h2 class="hasAnchor">
<a href="#learning-signatures" class="anchor"></a>Learning signatures</h2>
<p>For instance, the learning method <code>"PositiveProportionDifference"</code> requires prior clustering information to be stored as a <code>factor</code> in a column of the <code>colData</code> slot. The method can then be applied to identify markers for each cluster, using a variety of filters on individual markers (e.g., minimal difference in detection rate between the target cluster and any other cluster) and on the combined set of markers (e.g., minimal proportion of cells in the target cluster where all markers are detected simultaneously). Here, we use the Cre line information to find genes markers of the three populations of neurons.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">allen<span class="op">$</span>driver_<span class="dv">1</span>_s &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">as.factor</a></span>(allen<span class="op">$</span>driver_<span class="dv">1</span>_s)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">baseset &lt;-<span class="st"> </span><span class="kw"><a href="../reference/learnSignatures.html">learnSignatures</a></span>(</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">    <span class="dt">se =</span> allen, <span class="dt">assay.type =</span> <span class="st">"tophat_counts"</span>,</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">    <span class="dt">method =</span> <span class="st">"PositiveProportionDifference"</span>, <span class="dt">cluster.col =</span> <span class="st">"driver_1_s"</span>,</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">    <span class="dt">min.prop =</span> <span class="fl">0.5</span>, <span class="dt">min.diff =</span> <span class="fl">0.3</span>,  <span class="dt">diff.method =</span> <span class="st">"min"</span>)</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">baseset</a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt; BaseSets with 31 relations between 31 elements and 3 sets</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt;      element             set                        relationData</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt;   &lt;IdVector&gt;      &lt;IdVector&gt;                         &lt;DataFrame&gt;</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt; 1      Foxp2 Ntsr1-Cre_GN220                 1:0.773684210526316</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt; 2       Rprm Ntsr1-Cre_GN220 0.974683544303797:0.627315123251166</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt; 3    Gadd45a Ntsr1-Cre_GN220 0.974683544303797:0.516788807461692</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#&gt; 4      Ddah1 Ntsr1-Cre_GN220 0.974683544303797:0.511047180667434</span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt; 5       Tle4 Ntsr1-Cre_GN220                 1:0.510526315789474</span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt;          ...             ...                                 ...</span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt; 1      Pvrl1  Scnn1a-Tg3-Cre  0.936363636363636:0.34688995215311</span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="co">#&gt; 2    S100a10  Scnn1a-Tg3-Cre 0.936363636363636:0.336363636363636</span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="co">#&gt; 3       Coch  Scnn1a-Tg3-Cre 0.954545454545455:0.317703349282297</span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="co">#&gt; 4      Dkkl1  Scnn1a-Tg3-Cre 0.872727272727273:0.309569377990431</span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"><span class="co">#&gt; 5      Rims3  Scnn1a-Tg3-Cre 0.981818181818182:0.302870813397129</span></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="co">#&gt;         elementData           setData</span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="co">#&gt;         &lt;DataFrame&gt;       &lt;DataFrame&gt;</span></a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="co">#&gt; 1 0.361477572559367 0.556962025316456</span></a>
<a class="sourceLine" id="cb4-24" data-line-number="24"><span class="co">#&gt; 2 0.437994722955145 0.556962025316456</span></a>
<a class="sourceLine" id="cb4-25" data-line-number="25"><span class="co">#&gt; 3 0.482849604221636 0.556962025316456</span></a>
<a class="sourceLine" id="cb4-26" data-line-number="26"><span class="co">#&gt; 4 0.562005277044855 0.556962025316456</span></a>
<a class="sourceLine" id="cb4-27" data-line-number="27"><span class="co">#&gt; 5 0.551451187335092 0.556962025316456</span></a>
<a class="sourceLine" id="cb4-28" data-line-number="28"><span class="co">#&gt;                 ...               ...</span></a>
<a class="sourceLine" id="cb4-29" data-line-number="29"><span class="co">#&gt; 1 0.596306068601583 0.545454545454545</span></a>
<a class="sourceLine" id="cb4-30" data-line-number="30"><span class="co">#&gt; 2 0.638522427440633 0.545454545454545</span></a>
<a class="sourceLine" id="cb4-31" data-line-number="31"><span class="co">#&gt; 3 0.635883905013193 0.545454545454545</span></a>
<a class="sourceLine" id="cb4-32" data-line-number="32"><span class="co">#&gt; 4 0.577836411609499 0.545454545454545</span></a>
<a class="sourceLine" id="cb4-33" data-line-number="33"><span class="co">#&gt; 5 0.720316622691293 0.545454545454545</span></a></code></pre></div>
<p>In <em><a href="https://github.com/kevinrue/hancock">hancock</a></em>, learning methods return <code>BaseSets</code> objects, defined in the <em><a href="https://github.com/kevinrue/unisets">unisets</a></em> package. This container stores relations between elements (e.g., genes) and sets (e.g., signatures), along with optional metadata associated with each relation. In the next section, we explore the various pieces of information populated by <code>method = "PositiveProportionDifference")</code>.</p>
</div>
<div id="visualise-learning" class="section level2">
<h2 class="hasAnchor">
<a href="#visualise-learning" class="anchor"></a>Visualizing learning outputs</h2>
<div id="relation-metadata" class="section level3">
<h3 class="hasAnchor">
<a href="#relation-metadata" class="anchor"></a>Relation metadata</h3>
<p>Notably, the metadata associated with each relation between a marker (<code>"element"</code>) and the signature (<code>"set"</code>) can be flattened in a <code>data.frame</code> format. Specifically, the <code>"PositiveProportionDifference"</code> describes two pieces of information:</p>
<ul>
<li>
<code>"ProportionPositive"</code>, the proportion of cells with detectable expression of the marker in the associated cluster</li>
<li>
<code>"minDifferenceProportion"</code>, the minimal difference betwween the detection rate in the associated cluster compared with that of each other cluster</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(baseset))</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="co">#&gt;   element             set ProportionPositive minDifferenceProportion</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; 1   Foxp2 Ntsr1-Cre_GN220          1.0000000               0.7736842</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt; 2    Rprm Ntsr1-Cre_GN220          0.9746835               0.6273151</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt; 3 Gadd45a Ntsr1-Cre_GN220          0.9746835               0.5167888</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt; 4   Ddah1 Ntsr1-Cre_GN220          0.9746835               0.5110472</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt; 5    Tle4 Ntsr1-Cre_GN220          1.0000000               0.5105263</span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt; 6 Dpy19l3 Ntsr1-Cre_GN220          0.9113924               0.4750288</span></a></code></pre></div>
</div>
<div id="marker-metadata" class="section level3">
<h3 class="hasAnchor">
<a href="#marker-metadata" class="anchor"></a>Marker metadata</h3>
<p>In addition, metadata associated with each unique marker–irrespective of its specific relationships with individual signature–are stored in the metadata columns of the <code>elementData</code> slot. Specifically, the <code>"PositiveProportionDifference"</code> describes <code>"ProportionPositive"</code>, the proportion of cells with detectable expression of the markers across the entire data set.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">mcols</span>(<span class="kw">elementData</span>(baseset))</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="co">#&gt; DataFrame with 31 rows and 1 column</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">#&gt;         ProportionPositive</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt;                  &lt;numeric&gt;</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt; Foxp2    0.361477572559367</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt; Rprm     0.437994722955145</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; Gadd45a  0.482849604221636</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt; Ddah1    0.562005277044855</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt; Tle4     0.551451187335092</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt; ...                    ...</span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt; Pvrl1    0.596306068601583</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt; S100a10  0.638522427440633</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt; Coch     0.635883905013193</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt; Dkkl1    0.577836411609499</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt; Rims3    0.720316622691293</span></a></code></pre></div>
</div>
<div id="set-metadata" class="section level3">
<h3 class="hasAnchor">
<a href="#set-metadata" class="anchor"></a>Set metadata</h3>
<p>Finally, metadata associated with each unique signature–irrespective of its specific relationships with individual markers–are stored in the metadata columns of the <code>setData</code> slot. Specifically, the <code>"PositiveProportionDifference"</code> describes <code>"ProportionPositive"</code>, the proportion of cells with detectable expression of all markers associated with each signature across the entire data set.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">mcols</span>(<span class="kw">setData</span>(baseset))</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co">#&gt; DataFrame with 3 rows and 1 column</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt;                 ProportionPositive</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt;                          &lt;numeric&gt;</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt; Ntsr1-Cre_GN220  0.556962025316456</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt; Rbp4-Cre_KL100   0.505263157894737</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt; Scnn1a-Tg3-Cre   0.545454545454545</span></a></code></pre></div>
</div>
</div>
<div id="predict-proportion-positive" class="section level2">
<h2 class="hasAnchor">
<a href="#predict-proportion-positive" class="anchor"></a>Applying signatures to predict labels</h2>
<p>Markers <a href="#learning-signatures">learned previously</a> may then be applied on the training data set itself, to annotate each cluster with its corresponding signature.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">allen.hancock &lt;-<span class="st"> </span><span class="kw"><a href="../reference/predictSignatures.html">predict</a></span>(</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">    baseset, allen, <span class="dt">assay.type =</span> <span class="st">"tophat_counts"</span>,</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">    <span class="dt">method =</span> <span class="st">"ProportionPositive"</span>, <span class="dt">cluster.col=</span><span class="st">"driver_1_s"</span>)</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">allen.hancock</a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="co">#&gt; class: SummarizedExperiment </span></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="co">#&gt; dim: 20908 379 </span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="co">#&gt; metadata(3): SuppInfo which_qc hancock</span></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="co">#&gt; assays(4): tophat_counts cufflinks_fpkm rsem_counts rsem_tpm</span></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="co">#&gt; rownames(20908): 0610007P14Rik 0610009B22Rik ... Zzef1 Zzz3</span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="co">#&gt; rowData names(0):</span></a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="co">#&gt; colnames(379): SRR2140028 SRR2140022 ... SRR2139341 SRR2139336</span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="co">#&gt; colData names(23): NREADS NALIGNED ... passes_qc_checks_s hancock</span></a></code></pre></div>
<p>In <em><a href="https://github.com/kevinrue/hancock">hancock</a></em>, the <code>predict</code> function populates:</p>
<ul>
<li>the <code>"hancock"</code> column of <code>colData(allen.hancock)</code>
</li>
<li>the <code>"hancock"</code> item of <code>metadata(allen.hancock)</code>
</li>
</ul>
<p>In the next section, we explore the various pieces of information populated by <code>method = "ProportionPositive"</code>.</p>
</div>
<div id="visualise-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#visualise-prediction" class="anchor"></a>Visualizing prediction outputs</h2>
<div id="predicted-cell-label" class="section level3">
<h3 class="hasAnchor">
<a href="#predicted-cell-label" class="anchor"></a>Predicted cell label</h3>
<p>The key output of every prediction method is the cell identity predicted for each cell in the object. All prediction methods store this information in <code>colData(sce)[["hancock"]][["prediction"]]</code>, or <code>sce$hancock$prediction</code>, in short.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/summary">summary</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(<span class="kw">colData</span>(allen.hancock)[[<span class="st">"hancock"</span>]]))</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co">#&gt;            prediction </span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="co">#&gt;  Ntsr1-Cre_GN220: 79  </span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#&gt;  Rbp4-Cre_KL100 :190  </span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="co">#&gt;  Scnn1a-Tg3-Cre :110</span></a></code></pre></div>
<p>The <code>metadata</code> slot is used to store some required information:</p>
<ul>
<li>
<code>"GeneSets"</code>, the signatures used in the prediction process</li>
<li>
<code>"method"</code>, the unique identifier of the prediction method</li>
<li>
<code>"packageVersion"</code>, the version of the <em><a href="https://github.com/kevinrue/hancock">hancock</a></em> package that performed the prediction</li>
</ul>
<p>Optional method-specific information may be added by each prediction method. For <code>method="ProportionPositive"</code>, those are:</p>
<ul>
<li>
<code>"ProportionPositiveByCluster"</code>, the proportion of cells in each cluster that express all markers in each signature</li>
<li>
<code>"TopSignatureByCluster"</code>, the most frequently detected signature in each cluster</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw">metadata</span>(allen.hancock)[[<span class="st">"hancock"</span>]]</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co">#&gt; $GeneSets</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="co">#&gt; BaseSets with 31 relations between 31 elements and 3 sets</span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co">#&gt;      element             set                        relationData</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="co">#&gt;   &lt;IdVector&gt;      &lt;IdVector&gt;                         &lt;DataFrame&gt;</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="co">#&gt; 1      Foxp2 Ntsr1-Cre_GN220                 1:0.773684210526316</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="co">#&gt; 2       Rprm Ntsr1-Cre_GN220 0.974683544303797:0.627315123251166</span></a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="co">#&gt; 3    Gadd45a Ntsr1-Cre_GN220 0.974683544303797:0.516788807461692</span></a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="co">#&gt; 4      Ddah1 Ntsr1-Cre_GN220 0.974683544303797:0.511047180667434</span></a>
<a class="sourceLine" id="cb10-10" data-line-number="10"><span class="co">#&gt; 5       Tle4 Ntsr1-Cre_GN220                 1:0.510526315789474</span></a>
<a class="sourceLine" id="cb10-11" data-line-number="11"><span class="co">#&gt;          ...             ...                                 ...</span></a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="co">#&gt; 1      Pvrl1  Scnn1a-Tg3-Cre  0.936363636363636:0.34688995215311</span></a>
<a class="sourceLine" id="cb10-13" data-line-number="13"><span class="co">#&gt; 2    S100a10  Scnn1a-Tg3-Cre 0.936363636363636:0.336363636363636</span></a>
<a class="sourceLine" id="cb10-14" data-line-number="14"><span class="co">#&gt; 3       Coch  Scnn1a-Tg3-Cre 0.954545454545455:0.317703349282297</span></a>
<a class="sourceLine" id="cb10-15" data-line-number="15"><span class="co">#&gt; 4      Dkkl1  Scnn1a-Tg3-Cre 0.872727272727273:0.309569377990431</span></a>
<a class="sourceLine" id="cb10-16" data-line-number="16"><span class="co">#&gt; 5      Rims3  Scnn1a-Tg3-Cre 0.981818181818182:0.302870813397129</span></a>
<a class="sourceLine" id="cb10-17" data-line-number="17"><span class="co">#&gt;         elementData           setData</span></a>
<a class="sourceLine" id="cb10-18" data-line-number="18"><span class="co">#&gt;         &lt;DataFrame&gt;       &lt;DataFrame&gt;</span></a>
<a class="sourceLine" id="cb10-19" data-line-number="19"><span class="co">#&gt; 1 0.361477572559367 0.556962025316456</span></a>
<a class="sourceLine" id="cb10-20" data-line-number="20"><span class="co">#&gt; 2 0.437994722955145 0.556962025316456</span></a>
<a class="sourceLine" id="cb10-21" data-line-number="21"><span class="co">#&gt; 3 0.482849604221636 0.556962025316456</span></a>
<a class="sourceLine" id="cb10-22" data-line-number="22"><span class="co">#&gt; 4 0.562005277044855 0.556962025316456</span></a>
<a class="sourceLine" id="cb10-23" data-line-number="23"><span class="co">#&gt; 5 0.551451187335092 0.556962025316456</span></a>
<a class="sourceLine" id="cb10-24" data-line-number="24"><span class="co">#&gt;                 ...               ...</span></a>
<a class="sourceLine" id="cb10-25" data-line-number="25"><span class="co">#&gt; 1 0.596306068601583 0.545454545454545</span></a>
<a class="sourceLine" id="cb10-26" data-line-number="26"><span class="co">#&gt; 2 0.638522427440633 0.545454545454545</span></a>
<a class="sourceLine" id="cb10-27" data-line-number="27"><span class="co">#&gt; 3 0.635883905013193 0.545454545454545</span></a>
<a class="sourceLine" id="cb10-28" data-line-number="28"><span class="co">#&gt; 4 0.577836411609499 0.545454545454545</span></a>
<a class="sourceLine" id="cb10-29" data-line-number="29"><span class="co">#&gt; 5 0.720316622691293 0.545454545454545</span></a>
<a class="sourceLine" id="cb10-30" data-line-number="30"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb10-31" data-line-number="31"><span class="co">#&gt; $method</span></a>
<a class="sourceLine" id="cb10-32" data-line-number="32"><span class="co">#&gt; [1] "ProportionPositive"</span></a>
<a class="sourceLine" id="cb10-33" data-line-number="33"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb10-34" data-line-number="34"><span class="co">#&gt; $packageVersion</span></a>
<a class="sourceLine" id="cb10-35" data-line-number="35"><span class="co">#&gt; [1] '0.99.0'</span></a>
<a class="sourceLine" id="cb10-36" data-line-number="36"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb10-37" data-line-number="37"><span class="co">#&gt; $ProportionPositiveByCluster</span></a>
<a class="sourceLine" id="cb10-38" data-line-number="38"><span class="co">#&gt;                  signature</span></a>
<a class="sourceLine" id="cb10-39" data-line-number="39"><span class="co">#&gt; cluster           Ntsr1-Cre_GN220 Rbp4-Cre_KL100 Scnn1a-Tg3-Cre</span></a>
<a class="sourceLine" id="cb10-40" data-line-number="40"><span class="co">#&gt;   Ntsr1-Cre_GN220     0.556962025     0.05063291     0.00000000</span></a>
<a class="sourceLine" id="cb10-41" data-line-number="41"><span class="co">#&gt;   Rbp4-Cre_KL100      0.010526316     0.50526316     0.08421053</span></a>
<a class="sourceLine" id="cb10-42" data-line-number="42"><span class="co">#&gt;   Scnn1a-Tg3-Cre      0.009090909     0.03636364     0.54545455</span></a>
<a class="sourceLine" id="cb10-43" data-line-number="43"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb10-44" data-line-number="44"><span class="co">#&gt; $TopSignatureByCluster</span></a>
<a class="sourceLine" id="cb10-45" data-line-number="45"><span class="co">#&gt; Ntsr1-Cre_GN220  Rbp4-Cre_KL100  Scnn1a-Tg3-Cre </span></a>
<a class="sourceLine" id="cb10-46" data-line-number="46"><span class="co">#&gt; Ntsr1-Cre_GN220  Rbp4-Cre_KL100  Scnn1a-Tg3-Cre </span></a>
<a class="sourceLine" id="cb10-47" data-line-number="47"><span class="co">#&gt; Levels: Ntsr1-Cre_GN220 Rbp4-Cre_KL100 Scnn1a-Tg3-Cre</span></a></code></pre></div>
<p>In particular, <code>"ProportionPositiveByCluster"</code> may be visualized as a heat map using the <code>plotProportionPositive</code> method. This view is useful to examine the specificity of each signature for each cluster.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw"><a href="../reference/predictByProportionPositive.html">plotProportionPositive</a></span>(allen.hancock)</a></code></pre></div>
<p><img src="hancock_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
</div>
</div>
<div id="renaming-signatures" class="section level2">
<h2 class="hasAnchor">
<a href="#renaming-signatures" class="anchor"></a>Renaming signatures</h2>
<div id="manually" class="section level3">
<h3 class="hasAnchor">
<a href="#manually" class="anchor"></a>Manually</h3>
<p>Renaming a set of signatures is as simple as renaming the identifiers of the <code>setData</code> slot that stores the signatures.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">ids</span>(<span class="kw">setData</span>(baseset)) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">gsub</a></span>(<span class="st">"_"</span>, <span class="st">" "</span>, <span class="kw">ids</span>(<span class="kw">setData</span>(baseset)))</a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(<span class="kw">ids</span>(<span class="kw">sets</span>(baseset)))</a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="co">#&gt; Ntsr1-Cre GN220  Rbp4-Cre KL100  Scnn1a-Tg3-Cre </span></a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="co">#&gt;              15               6              10</span></a></code></pre></div>
</div>
</div>
<div id="interactively" class="section level2">
<h2 class="hasAnchor">
<a href="#interactively" class="anchor"></a>Interactively</h2>
<p>In addition, the <em><a href="https://github.com/kevinrue/hancock">hancock</a></em> package includes a lightweight <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> app that offers users the possibility to interactively rename signatures while inspecting their features in a <code>SummarizedExperiment</code> object (e.g., count of cells associated with each signature, layout in reduced dimension).</p>
<p>Specifically, the app requires a set of signatures and a <code>SummarizedExperiment</code> object that was previously annotated with those signatures using the <code>predict</code> function. When closed, the app returns the updated set of signatures.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">allen.hancock &lt;-<span class="st"> </span><span class="kw"><a href="../reference/predictSignatures.html">predict</a></span>(</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">    baseset, allen, <span class="dt">assay.type =</span> <span class="st">"tophat_counts"</span>, <span class="dt">method =</span> <span class="st">"ProportionPositive"</span>,</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">    <span class="dt">cluster.col=</span><span class="st">"driver_1_s"</span>)</a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb13-5" data-line-number="5">    <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(shiny)</a>
<a class="sourceLine" id="cb13-6" data-line-number="6">    baseset &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/shiny/topics/runApp">runApp</a></span>(<span class="kw"><a href="../reference/shinyLabels.html">shinyLabels</a></span>(baseset, allen.hancock))</a>
<a class="sourceLine" id="cb13-7" data-line-number="7">}</a></code></pre></div>
<p>Notably, this <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> app automatically detects the presence of optional dimensionality reduction results in <em><a href="https://bioconductor.org/packages/3.9/SingleCellExperiment">SingleCellExperiment</a></em> objects, allowing inspection and annotation of the gene signatures using that information.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(SingleCellExperiment)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scater)</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">allen.hancock &lt;-<span class="st"> </span><span class="kw">as</span>(allen.hancock, <span class="st">"SingleCellExperiment"</span>)</a>
<a class="sourceLine" id="cb14-4" data-line-number="4">allen.hancock &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/scater/topics/normalize">normalize</a></span>(allen.hancock, <span class="dt">exprs_values=</span><span class="st">"tophat_counts"</span>)</a>
<a class="sourceLine" id="cb14-5" data-line-number="5">allen.hancock &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/scater/topics/runPCA">runPCA</a></span>(allen.hancock)</a>
<a class="sourceLine" id="cb14-6" data-line-number="6">allen.hancock &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/scater/topics/runTSNE">runTSNE</a></span>(allen.hancock)</a>
<a class="sourceLine" id="cb14-7" data-line-number="7"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb14-8" data-line-number="8">    <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(shiny)</a>
<a class="sourceLine" id="cb14-9" data-line-number="9">    baseset &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/shiny/topics/runApp">runApp</a></span>(<span class="kw"><a href="../reference/shinyLabels.html">shinyLabels</a></span>(baseset, allen.hancock))</a>
<a class="sourceLine" id="cb14-10" data-line-number="10">}</a></code></pre></div>
</div>
</div>
<div id="types-of-signatures" class="section level1">
<h1 class="hasAnchor">
<a href="#types-of-signatures" class="anchor"></a>Types of signatures</h1>
<div id="absolute-markers" class="section level2">
<h2 class="hasAnchor">
<a href="#absolute-markers" class="anchor"></a>Absolute markers</h2>
<p>As described in the accompanying <a href="concepts.html">concepts</a> vignette, absolute markers (also known as “pan markers”) may be defined as genes detected in each cluster, irrespective of their expression in the other clusters.</p>
<p>For instance, the <code>"PositiveProportionDifference"</code> learning method can be used to identify such markers, by setting <code>min.diff=0</code> to annul any comparison between the detection frequency in the target cluster and all other clusters.</p>
<p>As the numbers of genes detected in each cluster may be rather large, it is generally a good idea to restrict markers to be detected in a very high fraction of the corresponding cluster, for instance <code>min.prop = 0.9</code>. In addition, the <code>threshold</code> argument may be used to define a minimal threshold of expression level to consider a marker as “detected” in each cell, and the <code>assay.type</code> argument declares the assay to use (e.g., “counts”, “logcouts”, “TPM”).</p>
<p>This ensures that for each cluster, the combined set of markers is simultaneously detected above 1 transcript per million (TPM) in at least 90% of cells in that cluster.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">baseset &lt;-<span class="st"> </span><span class="kw"><a href="../reference/learnSignatures.html">learnSignatures</a></span>(</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">    <span class="dt">se =</span> allen, <span class="dt">assay.type =</span> <span class="st">"rsem_tpm"</span>,</a>
<a class="sourceLine" id="cb15-3" data-line-number="3">    <span class="dt">method =</span> <span class="st">"PositiveProportionDifference"</span>, <span class="dt">cluster.col =</span> <span class="st">"driver_1_s"</span>,</a>
<a class="sourceLine" id="cb15-4" data-line-number="4">    <span class="dt">min.diff =</span> <span class="dv">0</span>, <span class="dt">min.prop =</span> <span class="fl">0.9</span>, <span class="dt">threshold =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb15-5" data-line-number="5"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(<span class="kw">ids</span>(<span class="kw">sets</span>(baseset)))</a>
<a class="sourceLine" id="cb15-6" data-line-number="6"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb15-7" data-line-number="7"><span class="co">#&gt; Ntsr1-Cre_GN220  Rbp4-Cre_KL100  Scnn1a-Tg3-Cre </span></a>
<a class="sourceLine" id="cb15-8" data-line-number="8"><span class="co">#&gt;             551             628             507</span></a>
<a class="sourceLine" id="cb15-9" data-line-number="9"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(baseset))</a>
<a class="sourceLine" id="cb15-10" data-line-number="10"><span class="co">#&gt;   element             set ProportionPositive minDifferenceProportion</span></a>
<a class="sourceLine" id="cb15-11" data-line-number="11"><span class="co">#&gt; 1   Foxp2 Ntsr1-Cre_GN220          1.0000000              0.91578947</span></a>
<a class="sourceLine" id="cb15-12" data-line-number="12"><span class="co">#&gt; 2    Nfia Ntsr1-Cre_GN220          1.0000000              0.47894737</span></a>
<a class="sourceLine" id="cb15-13" data-line-number="13"><span class="co">#&gt; 3 Sparcl1 Ntsr1-Cre_GN220          1.0000000              0.13684211</span></a>
<a class="sourceLine" id="cb15-14" data-line-number="14"><span class="co">#&gt; 4   Med10 Ntsr1-Cre_GN220          1.0000000              0.12631579</span></a>
<a class="sourceLine" id="cb15-15" data-line-number="15"><span class="co">#&gt; 5   Cplx2 Ntsr1-Cre_GN220          1.0000000              0.10000000</span></a>
<a class="sourceLine" id="cb15-16" data-line-number="16"><span class="co">#&gt; 6   Efhd2 Ntsr1-Cre_GN220          0.9873418              0.09260493</span></a></code></pre></div>
</div>
<div id="relative-markers" class="section level2">
<h2 class="hasAnchor">
<a href="#relative-markers" class="anchor"></a>Relative markers</h2>
<p>As described in the accompanying <a href="concepts.html">concepts</a> vignette, relative markers (also known as “key markers”) may be defined by <em>differential</em> analysis against other cells in the same sample.</p>
<p>For instance, the <code>"PositiveProportionDifference"</code> learning method can be used to identify such markers, by setting <code>min.diff</code> to a value greater than 0, in order to subset candidate markers to those detected at a rate at least 50% higher than the detection rate observed in any other cluster.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">baseset &lt;-<span class="st"> </span><span class="kw"><a href="../reference/learnSignatures.html">learnSignatures</a></span>(</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">    <span class="dt">se =</span> allen, <span class="dt">assay.type =</span> <span class="st">"tophat_counts"</span>,</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">    <span class="dt">method =</span> <span class="st">"PositiveProportionDifference"</span>, <span class="dt">cluster.col =</span> <span class="st">"driver_1_s"</span>,</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">    <span class="dt">min.diff =</span> <span class="fl">0.5</span>, <span class="dt">diff.method =</span> <span class="st">"min"</span>)</a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(<span class="kw">ids</span>(<span class="kw">sets</span>(baseset)))</a>
<a class="sourceLine" id="cb16-6" data-line-number="6"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-7" data-line-number="7"><span class="co">#&gt; Ntsr1-Cre_GN220  Rbp4-Cre_KL100  Scnn1a-Tg3-Cre </span></a>
<a class="sourceLine" id="cb16-8" data-line-number="8"><span class="co">#&gt;              14               2               4</span></a></code></pre></div>
<p>Among the learning outputs stored in the <code>BaseSets</code> metadata information, the relation metadata column <code>"minDifferenceProportion"</code> reflects the <code>min.diff=0.5</code> threshold applied when learning the signatures.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/summary">summary</a></span>(<span class="kw">mcols</span>(<span class="kw">relations</span>(baseset))[[<span class="st">"minDifferenceProportion"</span>]])</a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="co">#&gt;  0.5085  0.5341  0.5861  0.5949  0.6298  0.7737</span></a></code></pre></div>
<p>In addition, information about individual markers may be stored as element metadata accessible using the <code>elementData</code> and <code>mcols</code> methods as shown below. For instance, the proportion of cells positive for each marker across the entire data set.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw">mcols</span>(<span class="kw">elementData</span>(baseset))</a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="co">#&gt; DataFrame with 20 rows and 1 column</span></a>
<a class="sourceLine" id="cb18-3" data-line-number="3"><span class="co">#&gt;               ProportionPositive</span></a>
<a class="sourceLine" id="cb18-4" data-line-number="4"><span class="co">#&gt;                        &lt;numeric&gt;</span></a>
<a class="sourceLine" id="cb18-5" data-line-number="5"><span class="co">#&gt; Foxp2          0.361477572559367</span></a>
<a class="sourceLine" id="cb18-6" data-line-number="6"><span class="co">#&gt; Syt6           0.168865435356201</span></a>
<a class="sourceLine" id="cb18-7" data-line-number="7"><span class="co">#&gt; Thsd7b         0.266490765171504</span></a>
<a class="sourceLine" id="cb18-8" data-line-number="8"><span class="co">#&gt; Ifitm2         0.253298153034301</span></a>
<a class="sourceLine" id="cb18-9" data-line-number="9"><span class="co">#&gt; Rprm           0.437994722955145</span></a>
<a class="sourceLine" id="cb18-10" data-line-number="10"><span class="co">#&gt; ...                          ...</span></a>
<a class="sourceLine" id="cb18-11" data-line-number="11"><span class="co">#&gt; 6330403A02Rik  0.390501319261214</span></a>
<a class="sourceLine" id="cb18-12" data-line-number="12"><span class="co">#&gt; Scnn1a         0.263852242744063</span></a>
<a class="sourceLine" id="cb18-13" data-line-number="13"><span class="co">#&gt; Whrn           0.308707124010554</span></a>
<a class="sourceLine" id="cb18-14" data-line-number="14"><span class="co">#&gt; Igfn1          0.403693931398417</span></a>
<a class="sourceLine" id="cb18-15" data-line-number="15"><span class="co">#&gt; Grik1           0.33245382585752</span></a></code></pre></div>
<p>Similarly, information about individual sets may be stored as set metadata accessible using the <code>setData</code> and <code>mcols</code> methods as shown below. For instance, the proportion of cells simultaneously positive for all markers in the cluster where this signature was defined.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">mcols</span>(<span class="kw">setData</span>(baseset))</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="co">#&gt; DataFrame with 3 rows and 1 column</span></a>
<a class="sourceLine" id="cb19-3" data-line-number="3"><span class="co">#&gt;                 ProportionPositive</span></a>
<a class="sourceLine" id="cb19-4" data-line-number="4"><span class="co">#&gt;                          &lt;numeric&gt;</span></a>
<a class="sourceLine" id="cb19-5" data-line-number="5"><span class="co">#&gt; Ntsr1-Cre_GN220  0.113924050632911</span></a>
<a class="sourceLine" id="cb19-6" data-line-number="6"><span class="co">#&gt; Rbp4-Cre_KL100   0.642105263157895</span></a>
<a class="sourceLine" id="cb19-7" data-line-number="7"><span class="co">#&gt; Scnn1a-Tg3-Cre   0.418181818181818</span></a></code></pre></div>
<p>For instance, future methods to identify absolute markers could include differential expression between the target cluster and all other clusters to identify candidate markers significantly differentially expressed between clusters.</p>
</div>
</div>
<div id="additional-information" class="section level1">
<h1 class="hasAnchor">
<a href="#additional-information" class="anchor"></a>Additional information</h1>
<p>Bug reports can be posted as issues in the <em><a href="https://github.com/kevinrue/hancock">hancock</a></em> GitHub repository. The GitHub repository is the primary source for development versions of the package, where new functionality is added over time. The authors appreciate well-considered suggestions for improvements or new features, or even better, pull requests.</p>
<p>If you use <em><a href="https://github.com/kevinrue/hancock">hancock</a></em> for your analysis, please cite it as shown below:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/citation">citation</a></span>(<span class="st">"hancock"</span>)</a>
<a class="sourceLine" id="cb20-2" data-line-number="2"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb20-3" data-line-number="3"><span class="co">#&gt; To cite package 'hancock' in publications use:</span></a>
<a class="sourceLine" id="cb20-4" data-line-number="4"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb20-5" data-line-number="5"><span class="co">#&gt;   Kevin Rue-Albrecht, Robert Amezquita and Aaron Lun (2018).</span></a>
<a class="sourceLine" id="cb20-6" data-line-number="6"><span class="co">#&gt;   hancock: Learn and Apply Cell Type Signatures.</span></a>
<a class="sourceLine" id="cb20-7" data-line-number="7"><span class="co">#&gt;   https://github.com/kevinrue/hancock,</span></a>
<a class="sourceLine" id="cb20-8" data-line-number="8"><span class="co">#&gt;   http://kevinrue.github.io/hancock.</span></a>
<a class="sourceLine" id="cb20-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb20-10" data-line-number="10"><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></a>
<a class="sourceLine" id="cb20-11" data-line-number="11"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb20-12" data-line-number="12"><span class="co">#&gt;   @Manual{,</span></a>
<a class="sourceLine" id="cb20-13" data-line-number="13"><span class="co">#&gt;     title = {hancock: Learn and Apply Cell Type Signatures},</span></a>
<a class="sourceLine" id="cb20-14" data-line-number="14"><span class="co">#&gt;     author = {Kevin Rue-Albrecht and Robert Amezquita and Aaron Lun},</span></a>
<a class="sourceLine" id="cb20-15" data-line-number="15"><span class="co">#&gt;     year = {2018},</span></a>
<a class="sourceLine" id="cb20-16" data-line-number="16"><span class="co">#&gt;     note = {https://github.com/kevinrue/hancock,</span></a>
<a class="sourceLine" id="cb20-17" data-line-number="17"><span class="co">#&gt; http://kevinrue.github.io/hancock},</span></a>
<a class="sourceLine" id="cb20-18" data-line-number="18"><span class="co">#&gt;   }</span></a></code></pre></div>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/sessionInfo">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb21-2" data-line-number="2"><span class="co">#&gt; R Under development (unstable) (2019-02-15 r76105)</span></a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></a>
<a class="sourceLine" id="cb21-4" data-line-number="4"><span class="co">#&gt; Running under: Ubuntu 14.04.5 LTS</span></a>
<a class="sourceLine" id="cb21-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb21-7" data-line-number="7"><span class="co">#&gt; BLAS: /home/travis/R-bin/lib/R/lib/libRblas.so</span></a>
<a class="sourceLine" id="cb21-8" data-line-number="8"><span class="co">#&gt; LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so</span></a>
<a class="sourceLine" id="cb21-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-10" data-line-number="10"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb21-11" data-line-number="11"><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></a>
<a class="sourceLine" id="cb21-12" data-line-number="12"><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></a>
<a class="sourceLine" id="cb21-13" data-line-number="13"><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></a>
<a class="sourceLine" id="cb21-14" data-line-number="14"><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></a>
<a class="sourceLine" id="cb21-15" data-line-number="15"><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></a>
<a class="sourceLine" id="cb21-16" data-line-number="16"><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></a>
<a class="sourceLine" id="cb21-17" data-line-number="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-18" data-line-number="18"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb21-19" data-line-number="19"><span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span></a>
<a class="sourceLine" id="cb21-20" data-line-number="20"><span class="co">#&gt; [8] methods   base     </span></a>
<a class="sourceLine" id="cb21-21" data-line-number="21"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-22" data-line-number="22"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb21-23" data-line-number="23"><span class="co">#&gt;  [1] scater_1.11.10              ggplot2_3.1.0              </span></a>
<a class="sourceLine" id="cb21-24" data-line-number="24"><span class="co">#&gt;  [3] SingleCellExperiment_1.5.2  scRNAseq_1.9.0             </span></a>
<a class="sourceLine" id="cb21-25" data-line-number="25"><span class="co">#&gt;  [5] hancock_0.99.0              unisets_0.99.0             </span></a>
<a class="sourceLine" id="cb21-26" data-line-number="26"><span class="co">#&gt;  [7] SummarizedExperiment_1.13.0 DelayedArray_0.9.8         </span></a>
<a class="sourceLine" id="cb21-27" data-line-number="27"><span class="co">#&gt;  [9] BiocParallel_1.17.13        matrixStats_0.54.0         </span></a>
<a class="sourceLine" id="cb21-28" data-line-number="28"><span class="co">#&gt; [11] Biobase_2.43.1              GenomicRanges_1.35.1       </span></a>
<a class="sourceLine" id="cb21-29" data-line-number="29"><span class="co">#&gt; [13] GenomeInfoDb_1.19.2         IRanges_2.17.4             </span></a>
<a class="sourceLine" id="cb21-30" data-line-number="30"><span class="co">#&gt; [15] S4Vectors_0.21.10           BiocGenerics_0.29.1        </span></a>
<a class="sourceLine" id="cb21-31" data-line-number="31"><span class="co">#&gt; [17] BiocStyle_2.11.0           </span></a>
<a class="sourceLine" id="cb21-32" data-line-number="32"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-33" data-line-number="33"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb21-34" data-line-number="34"><span class="co">#&gt;  [1] bitops_1.0-6             fs_1.2.6                </span></a>
<a class="sourceLine" id="cb21-35" data-line-number="35"><span class="co">#&gt;  [3] bit64_0.9-7              RColorBrewer_1.1-2      </span></a>
<a class="sourceLine" id="cb21-36" data-line-number="36"><span class="co">#&gt;  [5] rprojroot_1.3-2          tools_3.6.0             </span></a>
<a class="sourceLine" id="cb21-37" data-line-number="37"><span class="co">#&gt;  [7] backports_1.1.2          R6_2.4.0                </span></a>
<a class="sourceLine" id="cb21-38" data-line-number="38"><span class="co">#&gt;  [9] vipor_0.4.5              DBI_1.0.0               </span></a>
<a class="sourceLine" id="cb21-39" data-line-number="39"><span class="co">#&gt; [11] lazyeval_0.2.1           colorspace_1.4-0        </span></a>
<a class="sourceLine" id="cb21-40" data-line-number="40"><span class="co">#&gt; [13] GetoptLong_0.1.7         withr_2.1.2             </span></a>
<a class="sourceLine" id="cb21-41" data-line-number="41"><span class="co">#&gt; [15] gridExtra_2.3            tidyselect_0.2.5        </span></a>
<a class="sourceLine" id="cb21-42" data-line-number="42"><span class="co">#&gt; [17] bit_1.1-14               compiler_3.6.0          </span></a>
<a class="sourceLine" id="cb21-43" data-line-number="43"><span class="co">#&gt; [19] graph_1.61.0             BiocNeighbors_1.1.11    </span></a>
<a class="sourceLine" id="cb21-44" data-line-number="44"><span class="co">#&gt; [21] xml2_1.2.0               desc_1.2.0              </span></a>
<a class="sourceLine" id="cb21-45" data-line-number="45"><span class="co">#&gt; [23] rtracklayer_1.43.1       bookdown_0.9            </span></a>
<a class="sourceLine" id="cb21-46" data-line-number="46"><span class="co">#&gt; [25] scales_1.0.0             pkgdown_1.3.0           </span></a>
<a class="sourceLine" id="cb21-47" data-line-number="47"><span class="co">#&gt; [27] commonmark_1.7           stringr_1.4.0           </span></a>
<a class="sourceLine" id="cb21-48" data-line-number="48"><span class="co">#&gt; [29] digest_0.6.18            Rsamtools_1.99.1        </span></a>
<a class="sourceLine" id="cb21-49" data-line-number="49"><span class="co">#&gt; [31] rmarkdown_1.11           XVector_0.23.0          </span></a>
<a class="sourceLine" id="cb21-50" data-line-number="50"><span class="co">#&gt; [33] pkgconfig_2.0.2          htmltools_0.3.6         </span></a>
<a class="sourceLine" id="cb21-51" data-line-number="51"><span class="co">#&gt; [35] rlang_0.3.1              GlobalOptions_0.1.0     </span></a>
<a class="sourceLine" id="cb21-52" data-line-number="52"><span class="co">#&gt; [37] RSQLite_2.1.1            DelayedMatrixStats_1.5.2</span></a>
<a class="sourceLine" id="cb21-53" data-line-number="53"><span class="co">#&gt; [39] shiny_1.2.0              shape_1.4.4             </span></a>
<a class="sourceLine" id="cb21-54" data-line-number="54"><span class="co">#&gt; [41] jsonlite_1.6             dplyr_0.8.0             </span></a>
<a class="sourceLine" id="cb21-55" data-line-number="55"><span class="co">#&gt; [43] RCurl_1.95-4.11          magrittr_1.5            </span></a>
<a class="sourceLine" id="cb21-56" data-line-number="56"><span class="co">#&gt; [45] GenomeInfoDbData_1.2.0   Matrix_1.2-15           </span></a>
<a class="sourceLine" id="cb21-57" data-line-number="57"><span class="co">#&gt; [47] ggbeeswarm_0.6.0         Rcpp_1.0.0              </span></a>
<a class="sourceLine" id="cb21-58" data-line-number="58"><span class="co">#&gt; [49] munsell_0.5.0            GeneSet_0.0.1           </span></a>
<a class="sourceLine" id="cb21-59" data-line-number="59"><span class="co">#&gt; [51] viridis_0.5.1            rintrojs_0.2.0          </span></a>
<a class="sourceLine" id="cb21-60" data-line-number="60"><span class="co">#&gt; [53] stringi_1.3.1            yaml_2.2.0              </span></a>
<a class="sourceLine" id="cb21-61" data-line-number="61"><span class="co">#&gt; [55] MASS_7.3-51.1            zlibbioc_1.29.0         </span></a>
<a class="sourceLine" id="cb21-62" data-line-number="62"><span class="co">#&gt; [57] Rtsne_0.15               plyr_1.8.4              </span></a>
<a class="sourceLine" id="cb21-63" data-line-number="63"><span class="co">#&gt; [59] grid_3.6.0               blob_1.1.1              </span></a>
<a class="sourceLine" id="cb21-64" data-line-number="64"><span class="co">#&gt; [61] promises_1.0.1           shinydashboard_0.7.1    </span></a>
<a class="sourceLine" id="cb21-65" data-line-number="65"><span class="co">#&gt; [63] crayon_1.3.4             lattice_0.20-38         </span></a>
<a class="sourceLine" id="cb21-66" data-line-number="66"><span class="co">#&gt; [65] Biostrings_2.51.2        cowplot_0.9.4           </span></a>
<a class="sourceLine" id="cb21-67" data-line-number="67"><span class="co">#&gt; [67] annotate_1.61.0          circlize_0.4.5          </span></a>
<a class="sourceLine" id="cb21-68" data-line-number="68"><span class="co">#&gt; [69] knitr_1.21               ComplexHeatmap_1.99.5   </span></a>
<a class="sourceLine" id="cb21-69" data-line-number="69"><span class="co">#&gt; [71] pillar_1.3.1             rjson_0.2.20            </span></a>
<a class="sourceLine" id="cb21-70" data-line-number="70"><span class="co">#&gt; [73] reshape2_1.4.3           XML_3.98-1.17           </span></a>
<a class="sourceLine" id="cb21-71" data-line-number="71"><span class="co">#&gt; [75] glue_1.3.0               evaluate_0.13           </span></a>
<a class="sourceLine" id="cb21-72" data-line-number="72"><span class="co">#&gt; [77] BiocManager_1.30.4       httpuv_1.4.5.1          </span></a>
<a class="sourceLine" id="cb21-73" data-line-number="73"><span class="co">#&gt; [79] gtable_0.2.0             purrr_0.3.0             </span></a>
<a class="sourceLine" id="cb21-74" data-line-number="74"><span class="co">#&gt; [81] assertthat_0.2.0         xfun_0.4                </span></a>
<a class="sourceLine" id="cb21-75" data-line-number="75"><span class="co">#&gt; [83] mime_0.6                 xtable_1.8-3            </span></a>
<a class="sourceLine" id="cb21-76" data-line-number="76"><span class="co">#&gt; [85] later_0.8.0              roxygen2_6.1.1          </span></a>
<a class="sourceLine" id="cb21-77" data-line-number="77"><span class="co">#&gt; [87] viridisLite_0.3.0        tibble_2.0.1            </span></a>
<a class="sourceLine" id="cb21-78" data-line-number="78"><span class="co">#&gt; [89] beeswarm_0.2.3           GenomicAlignments_1.19.1</span></a>
<a class="sourceLine" id="cb21-79" data-line-number="79"><span class="co">#&gt; [91] AnnotationDbi_1.45.0     memoise_1.1.0           </span></a>
<a class="sourceLine" id="cb21-80" data-line-number="80"><span class="co">#&gt; [93] GSEABase_1.45.0</span></a>
<a class="sourceLine" id="cb21-81" data-line-number="81"><span class="co"># devtools::session_info()</span></a></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-tasic2016">
<p>Tasic, B., V. Menon, T. N. Nguyen, T. K. Kim, T. Jarsky, Z. Yao, B. Levi, et al. 2016. “Adult Mouse Cortical Cell Taxonomy Revealed by Single Cell Transcriptomics.” Journal Article. <em>Nat Neurosci</em> 19 (2): 335–46. <a href="https://doi.org/10.1038/nn.4216" class="uri">https://doi.org/10.1038/nn.4216</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#overview">Overview</a></li>
      <li>
<a href="#getting-started">Getting started</a><ul class="nav nav-pills nav-stacked">
<li><a href="#setup">Setup</a></li>
      <li><a href="#example-data-set">Example data set</a></li>
      <li><a href="#learning-signatures">Learning signatures</a></li>
      <li><a href="#visualise-learning">Visualizing learning outputs</a></li>
      <li><a href="#predict-proportion-positive">Applying signatures to predict labels</a></li>
      <li><a href="#visualise-prediction">Visualizing prediction outputs</a></li>
      <li><a href="#renaming-signatures">Renaming signatures</a></li>
      <li><a href="#interactively">Interactively</a></li>
      </ul>
</li>
      <li>
<a href="#types-of-signatures">Types of signatures</a><ul class="nav nav-pills nav-stacked">
<li><a href="#absolute-markers">Absolute markers</a></li>
      <li><a href="#relative-markers">Relative markers</a></li>
      </ul>
</li>
      <li><a href="#additional-information">Additional information</a></li>
      <li><a href="#session-info">Session Info</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Kevin Rue-Albrecht, Robert Amezquita, Aaron Lun.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
